### PR TITLE
GH-158 - Report the number of engagement reports submitted within a given time frame

### PR DESCRIPTION
**Pull request by evdhoudt**

### CODE REVIEW

**Comment by gjvoosten** [comment]
One commit probably refers to the wrong issue (should be #158).

File: src/main/java/mil/dds/anet/beans/search/ReportSearchQuery.java [hunk 1]
```diff
@@ -74,6 +76,22 @@ public void setEngagementDateEnd(DateTime engagementDateEnd) {
...
+	public Integer getEngagementDayOfWeek() {
+		return engagementDayOfWeek;
+	}
+
+	public void setEngagementDayOfWeek(Integer engagementDayOfWeek) {
+		this.engagementDayOfWeek = engagementDayOfWeek;
+	}
+
+	public boolean getIncludeEngagementDayOfWeek() {
+		return (includeEngagementDayOfWeek == null) ? false : includeEngagementDayOfWeek;
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/beans/search/ReportSearchQuery.java]: Better to use:
```		return Boolean.TRUE.equals(includeEngagementDayOfWeek);```
(should actually be changed elsewhere in the file as well).

File: src/main/java/mil/dds/anet/beans/search/ReportSearchQuery.java [hunk 2]
```diff
@@ -15,6 +15,8 @@
 	Integer authorId;
 	DateTime engagementDateStart;
 	DateTime engagementDateEnd;
+	Integer engagementDayOfWeek;
+	Boolean includeEngagementDayOfWeek;
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/beans/search/ReportSearchQuery.java]: I prefer to make new fields `final` (at some point [e.g. when we have decided on a Checkstyle profile] we should consider changing them all).

File: src/main/java/mil/dds/anet/beans/Report.java [hunk 3]
```diff
@@ -37,6 +37,7 @@
 	DateTime releasedAt;
 	
 	DateTime engagementDate;
+	Integer engagementDayOfWeek;
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/beans/Report.java]: I prefer to make new fields `final` (at some point [e.g. when we have decided on a Checkstyle profile] we should consider changing them all).

File: client/src/pages/insights/Show.js [hunk 4]
```diff
@@ -65,7 +92,45 @@ export default class InsightsShow extends Page {
...
+    })
+  }
+
+  updateDateRange = (value) => {
+    if (value.start !== null) {
+      this.updateDate("startDate", moment(value.start))
+    }
+
+    if (value.end !== null) {
+      this.updateDate("endDate", moment(value.end))
```
Comment (evdhoudt) [inline client/src/pages/insights/Show.js]: Fixed input to value.end

File: src/main/java/mil/dds/anet/beans/Report.java (hunk 3 above)
Comment (evdhoudt) [reply src/main/java/mil/dds/anet/beans/Report.java]: done

**Pull request merged**